src/xo-app/backup/restore/index.js

```diff
@@ -98,16 +102,53 @@ const doImport = ({ backup, sr, start }) => {
   }
 }
 
+class _Collapsible extends Component {
+  render () {
+    const {
+      collapsible,
+      children,
+      ...props
+    } = this.props
+
+    return collapsible
+      ? <Collapse {...props}> {children} </Collapse>
+      : <div> {children} </div>
+  }
+}
+
 class _ModalBody extends Component {
   get value () {
     return this.state
   }
 
+  _srPredicate = createSelector(
+    sr => sr,
+    () => this.state.sr.$pool,
+    (sr, pool) => isSrWritable(sr) && sr.$pool === pool
+  )
+
+  _onChange = event => {
+    const sr = getEventValue(event)
+
+    if (this.state.sr && this.state.sr.$pool !== sr.$pool) {
+      this.setState({
+        mapVdisSrs: undefined
+      })
+    }
+
+    this.setState({
+      sr
+    })
+  }
+
+  _getSelectedValue = vdiId => this.state.mapVdisSrs && this.state.mapVdisSrs[vdiId]
```

Comment:
Why don't you just use `this.state.mapVdisSrs[vdi.uuid]` in the map and always keep `mapVdisSrs` in the state? You could initialize the state with `{ mapVdisSrs: {} }` and call `this.setState({ mapVdisSrs: {} })` to reinitialize it, right?